### PR TITLE
DAH-2293 check for preferences existence

### PR DESCRIPTION
**Pull request by tallulahkay**
[DAH-2293](https://sfgovdt.jira.com/browse/DAH-2293)

Msol found a bug when testing. This fixes it.

[DAH-2293]: https://sfgovdt.jira.com/browse/DAH-2293?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/javascript/components/lease_ups/LeaseUpApplicationsPage.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ const getPageHeaderData = (listing, reportId) => {
 }
 
 const getPreferences = (listing) => {
-  if (!listing) return []
+  if (!listing || !listing.listing_lottery_preferences) return []
```
Comment (jimlin-sfgov) [inline app/javascript/components/lease_ups/LeaseUpApplicationsPage.js]: suggestion: replace with `if (!listing?.listing_lottery_preferences)`

File: app/javascript/components/supplemental_application/utils/supplementalApplicationUtils.js [hunk 2]
```diff
@@ -5,9 +5,11 @@ export const isLeaseAlreadyCreated = (lease) => lease && !!lease.id
...
 
 export const totalSetAsidesForPref = (listing, prefName) =>
-  listing.listing_lottery_preferences.find((pref) =>
-    pref.lottery_preference.name.includes(prefName)
-  )?.available_units ?? 0
+  listing.listing_lottery_preferences
+    ? listing.listing_lottery_preferences.find((pref) =>
+        pref.lottery_preference.name.includes(prefName)
+      )?.available_units ?? 0
+    : 0
```
Comment (jimlin-sfgov) [inline app/javascript/components/supplemental_application/utils/supplementalApplicationUtils.js:12]: suggestion: optional chaining should work here as well, something like `listing.listing_lottery_preferences?.find...`

**Review by jimlin-sfgov — APPROVED** [review]
LG

tested by followed msol's comments in the Jira ticket, but explicit testing instructions would be appreciated for next time

File: app/javascript/components/supplemental_application/utils/supplementalApplicationUtils.js (hunk 2 above)
Comment (tallulahkay) [reply app/javascript/components/supplemental_application/utils/supplementalApplicationUtils.js:12]: this one I tried, and it didn't work

**Pull request merged**